File: Seori/LeetCode 75/Hash Map_Set/1207. Unique Number of Occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        occurences = {}
        for num in arr:
            if occurences.get(num):
                occurences[num] += 1
            else:
                occurences[num] = 1

        occurences_values = occurences.values()
        return len(occurences_values) == len(set(occurences_values))

File: Gombang/[LeetCode75] - [Hash_Map_Set]/1207_Unique_Number_of_Occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int, int> map;
        for (int num : arr)
        {
            map[num]++;
        }

        vector<int> freqVec;
        for (auto it = map.begin(); it != map.end(); ++it)
        {
            freqVec.push_back(it->second);
        }

        unordered_set<int> uniqueSet(freqVec.begin(), freqVec.end());

        return freqVec.size() == uniqueSet.size();
    }
};